### PR TITLE
[libwebsockets] Enable separate debug and release builds

### PR DESCRIPTION
**Pull request by ahmedyarub**
Enable separate debug and release builds. The build used to fail if one of the target files did not exist since the portfile expected both of them to exist.

- #### What does your PR fix?  
  libwebsockets build

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all, Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/libwebsockets/portfile.cmake [hunk 1]
```diff
@@ -163,12 +163,18 @@ file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/share/libwebsockets-test-server")
...
+    file(READ "${CURRENT_PACKAGES_DIR}/share/libwebsockets/LibwebsocketsTargets-debug.cmake" LIBWEBSOCKETSTARGETSDEBUG_CMAKE)
+    string(REPLACE "websockets_static.lib" "websockets.lib" LIBWEBSOCKETSTARGETSDEBUG_CMAKE "${LIBWEBSOCKETSTARGETSDEBUG_CMAKE}")
+    file(WRITE "${CURRENT_PACKAGES_DIR}/share/libwebsockets/LibwebsocketsTargets-debug.cmake" "${LIBWEBSOCKETSTARGETSDEBUG_CMAKE}")
+endif()
+
+if(EXISTS "${CURRENT_PACKAGES_DIR}/share/libwebsockets/LibwebsocketsTargets-release.cmake")
+    file(READ "${CURRENT_PACKAGES_DIR}/share/libwebsockets/LibwebsocketsTargets-release.cmake" LIBWEBSOCKETSTARGETSRELEASE_CMAKE)
+    string(REPLACE "websockets_static.lib" "websockets.lib" LIBWEBSOCKETSTARGETSRELEASE_CMAKE "${LIBWEBSOCKETSTARGETSRELEASE_CMAKE}")
+    file(WRITE "${CURRENT_PACKAGES_DIR}/share/libwebsockets/LibwebsocketsTargets-release.cmake" "${LIBWEBSOCKETSTARGETSRELEASE_CMAKE}")
+endif()
```
Comment (dg0yt) [inline ports/libwebsockets/portfile.cmake:177]: a) I would recommend to test for build type, not file. It helps to notice mistyped file paths. (Just learned the hard way.)

~~~cmake
if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
~~~

b) There is a convenient `vcpkg_replace_string(<FILE> <PATTERN> <REPLACEMENT>)` function.

Comment (ahmedyarub) [reply ports/libwebsockets/portfile.cmake:173]: Done

**Review by PhoebeHui — APPROVED** [review]
Thanks for your PR!

**Comment by strega-nil-ms** [comment]
Thanks @ahmedyarub !

**Pull request merged**